### PR TITLE
DE-4537 Anomaly detection in cloudwatch

### PR DESCRIPTION
**Pull request by jmalyszko-fandom**
* query scanned bytes numbers are sent to the couldwatch
* terraform file to generate cloudwatch anomaly detection metrics 
* there are now two possible slack notifiers
* notifier.py reads from two different sources - additional one is the anomaly detection queue

### CODE REVIEW

File: bin/notification.py [hunk 1]
```diff
@@ -64,6 +87,60 @@ def send_slack_to_user(self, user_id, text):
...
+        message = AnomalyDetectionMessage(record_sns.Message)
+        new_state_value = message.new_state_value
+        if str.lower(new_state_value) == 'alarm':
+            new_state_reason = message.new_state_reason
+            user = message.user
+            params = dict(subject=subject, user=user, new_state_reason=new_state_reason)
+            text = self.config.SLACK_ANOMALY_DETECTION_THRESHOLD_MESSAGE.format(**params)
+            if hasattr(self.config, 'SLACK_USER_MAPPINGS'):
+                slack_user = self.config.SLACK_USER_MAPPINGS.get(user)
+            self.send_slack_to_channel(text)
```
Comment (mikulskibartosz) [inline bin/notification.py]: not very important, but can you move this line above the hasattr check? When you do it, we will have the code that sends messages to slack channel separated from the code that deals with sending messages to individual users

File: bin/usage_update.py [hunk 2]
```diff
@@ -11,27 +11,33 @@
...
 from query_dao import QueryDao
+from cloudwatch_metrics import CloudwatchMetrics
 from model import AthenaQuery, QueryState, TIMESTAMP_FORMAT
 
 
 def lambda_handler(event, context):
     athena = boto3.client('athena')
     sqs = boto3.client('sqs')
     dynamodb = boto3.resource('dynamodb')
+    if settings.USE_CLOUDWATCH_ANOMALY_DETECTION:
```
Comment (mikulskibartosz) [inline bin/usage_update.py]: you don't use this cloudwatch instance. CloudwatchMetrics.report_query_metric uses the one defined in the CloudwatchMetrics class

Comment (jmalyszko-fandom) [reply bin/usage_update.py]: good one :D

**Comment by jmalyszko-fandom** [comment]
Fixes pushed

**Review by mikulskibartosz — APPROVED** [review]

**Pull request merged**